src/cpu/minor/dyn_inst.cc

```diff
@@ -169,7 +174,7 @@ MinorDynInst::minorTraceInst(const Named &named_object) const
 {
     if (isFault()) {
         minorInst(named_object, "id=F;%s addr=0x%x fault=\"%s\"\n",
-            id, pc->instAddr(), fault->name());
+            id, pc ? pc->instAddr() : 0, fault->name());
```

Comment:
Shouldn't this be `pc->instAddr() ? pc->instAddr() : 0`? This change isn't just setting to 0 if `pc->instAddr` is NULL, it's setting to `pc` when it's not.

Reply:
No, the problem is that `pc` is was made into a `unique_ptr`, so you need to check to make sure that it's non-null before dereferencing.

The goal is to print "0" if `pc` is null (because we do not know what `pc` is) and print `pc->instAddr()` otherwise.

Reply:
Oh duh. I now get what my confusion is. I've been spending too much time as Python. I was parsing that condition as `pc if pc->instAdd() else 0`. 

Ok, This is all fine then.